internal/store/interfaces.go

```diff
@@ -8,37 +8,11 @@ import (
 
 //go:generate mockgen -source ./interfaces.go -destination ./mocks/interfaces.go -package mocks StatusTrackingGateway,Gateway,RouteTrackingListener,Listener,StatusTrackingRoute,Route,Store
 
-// StatusTrackingGateway is an optional extension
-// of Gateway. If supported by a Store, when
-// a Gateway is synced to an external location,
-// its corresponding callbacks should
-// be called.
-type StatusTrackingGateway interface {
-	Gateway
-
-	TrackSync(ctx context.Context, sync func() (bool, error)) error
-}
-
 // Gateway describes a gateway.
 type Gateway interface {
 	ID() core.GatewayID
-	Bind(ctx context.Context, route Route) []string
-	Remove(ctx context.Context, id string) error
 	Resolve() core.ResolvedGateway
-	CanFetchSecrets(ctx context.Context, secrets []string) (bool, error)
```

Comment:
Removed the `Context`/`error` since really the secret checks are a property of the `Gateway` itself. This also gives us an interface that reflects a bit more of what we want -- just a bag of structured data rather than something that's potentially holding on to things like kubernetes/consul clients it can use for resolving extra bits.